cfy_manager/components/usage_collector/scripts/collect_cloudify_uptime.py

```diff
@@ -1,19 +1,30 @@
 from script_utils import (logger,
                           send_data,
+                          HOURS_LOCK,
+                          HOURS_INTERVAL,
                           collect_metadata,
-                          create_manager_id_file)
+                          should_send_data,
+                          unlock_usage_collector,
+                          try_usage_collector_lock)
 
 
 CLOUDIFY_ENDPOINT_UPTIME_URL = 'https://api.cloudify.co/cloudifyUptime'
 
 
 def main():
-    logger.info('Uptime script started running')
-    create_manager_id_file()
-    data = {}
-    collect_metadata(data)
-    send_data(data, CLOUDIFY_ENDPOINT_UPTIME_URL)
-    logger.info('Uptime script finished running')
+    if try_usage_collector_lock(HOURS_LOCK):
+        logger.info('Acquired usage_collector table lock')
+        if should_send_data(HOURS_INTERVAL):
+            logger.info('Uptime script started running')
+            data = {}
+            collect_metadata(data)
+            send_data(data, CLOUDIFY_ENDPOINT_UPTIME_URL, HOURS_INTERVAL)
+            logger.info('Uptime script finished running')
+        else:
+            logger.info('cloudify_uptime was updated by a different Manager')
+        unlock_usage_collector(HOURS_LOCK)
```

Comment:
this really needs a `with` or a `try/finally`
i'm thinking like this:

```python
with usage_collector_lock(HOURS) as locked:
   if not locked:
       logger.info('other manager blabla')
       return
    etc
```
this of course needs an impl. of the lock:
```python
from contextlib import contextmanager
@contextmanager
def usage_collector_lock():
    locked = db.session.execute(...)
    try:
        yield locked
    finally:
        if locked:
            db.session.execute('unlock')
```

alternatively the `try/finally` approach is to just move it here, and do
```python
if try_usage_collector_lock(HOURS):
    try:
        collect()
        send()
    finally:
        unlock_usage_collector(HOURS)
````
the point is, if collect_metadata or send_data throws an exception (which it might, it's sending, ie. using the network!), then `unlock_usage_collector` will not be called. That's why things like releasing a lock must be in a `finally`

organizing the try/finally as a contextmanager has the advantage that it's impossible to forget to unlock, because the unlock happens automatically
but in this case, with the `if locked`, it is a bit weird, so up to you

Reply:
Great point, I'll change it.